### PR TITLE
Added a filter to our email admin toolie

### LINKED ISSUE #1967: Feature request: search function in the mail toolie
We now have 60 pages of emails. When I want to look through how we did for instance the camp emails of last year clicking through this is a hassle. It would be nice if you could search the descriptions with an optional checkbox also to search the body.

### PR DESCRIPTION
**Pull request by ysbrandB**
Fixes #1967.
Adds the ability to search for emails based on either internal description, email subject or body.
image

### CODE REVIEW

**Comment by Menkveld-24** [comment]
Ga slapen😂

File: app/Http/Controllers/EmailController.php [hunk 1]
```diff
@@ -27,6 +27,37 @@ public function index()
         ]);
     }
 
+    /** @return View */
+    public function filter(Request $request)
```
Comment (Menkveld-24) [inline app/Http/Controllers/EmailController.php:31]:
```suggestion
    public function filter(Request $request): View
```

File: app/Http/Controllers/EmailController.php [hunk 2]
```diff
@@ -27,6 +27,37 @@ public function index()
...
+            $filteredEmails = $filteredEmails->where('description', 'LIKE', '%'.$searchTerm.'%');
+        }
+
+        if ($subject) {
+            $filteredEmails = $filteredEmails->where('subject', 'LIKE', '%'.$searchTerm.'%');
+        }
+
+        if ($body) {
+            $filteredEmails = $filteredEmails->where('body', 'LIKE', '%'.$searchTerm.'%');
+        }
```
Comment (Menkveld-24) [inline app/Http/Controllers/EmailController.php:49]: Don't you overwrite the filteredEmails here so you practically can only use 1 filter at a time?

File: resources/views/emailadmin/admin_includes/filter.blade.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <div class="card mb-3">
+
+        <div class="card-header bg-dark text-white">
+            Search emails
+        </div>
+
+        <div class="card-body row">
+            <div class="form-group col-7">
+                <label for="searchterm">Searchterm:</label>
+                <input class="form-control" id="searchterm" name="searchterm" value="{{$searchTerm??""}}" required>
```
Comment (Menkveld-24) [inline resources/views/emailadmin/admin_includes/filter.blade.php:11]:
```suggestion
                <input class="form-control" id="searchterm" name="searchterm" value="{{ $searchTerm ?? "" }}" required>
```

File: resources/views/emailadmin/admin_includes/filter.blade.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            </div>
+
+            <div class="col">
+                @include('components.forms.checkbox',[
+                                    'name' => 'search_description',
+                                    'label' => 'Search description',
+                                    'checked' => $description??true,
```
Comment (Menkveld-24) [inline resources/views/emailadmin/admin_includes/filter.blade.php:18]:
```suggestion
                                    'checked' => $description ?? true,
```

File: resources/views/emailadmin/admin_includes/filter.blade.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                                ])
+                @include('components.forms.checkbox',[
+                                    'name' => 'search_subject',
+                                    'label' => 'Search subject',
+                                    'checked' => $subject??false,
```
Comment (Menkveld-24) [inline resources/views/emailadmin/admin_includes/filter.blade.php:23]:
```suggestion
                                    'checked' => $subject ?? false,
```

File: resources/views/emailadmin/admin_includes/filter.blade.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+                                ])
+                @include('components.forms.checkbox',[
+                                    'name' => 'search_body',
+                                    'label' => 'Search body',
+                                    'checked' => $body??false,
```
Comment (Menkveld-24) [inline resources/views/emailadmin/admin_includes/filter.blade.php:28]:
```suggestion
                                    'checked' => $body ?? false,
```

File: resources/views/emailadmin/overview.blade.php [hunk 7]
```diff
@@ -86,6 +86,13 @@
 
             </div>
 
+            @include('emailadmin.admin_includes.filter', [
+                'searchTerm' => $searchTerm??null,
+                'description' => $description??null,
+                'subject' => $subject??null,
+                'body' => $body??null,
```
Comment (Menkveld-24) [inline resources/views/emailadmin/overview.blade.php:93]:
```suggestion
                'searchTerm' => $searchTerm ?? null,
                'description' => $description ?? null,
                'subject' => $subject ?? null,
                'body' => $body ?? null,
```

**Review by Menkveld-24 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/EmailController.php (hunk 2 above)
Comment (ysbrandB) [reply app/Http/Controllers/EmailController.php]: Yes, you are right, meant to use an orWhere()

**Review by Menkveld-24 — APPROVED** [review]

**Pull request merged**